### PR TITLE
Add RTK endpoints for loading and searching parameters

### LINKED ISSUE #44290: Add RTK query endpoints for parameter values

### PR DESCRIPTION
**Pull request by romeovs**
Closes https://github.com/metabase/metabase/issues/44290

### Description

Adds RTK query endpoints to search and fetch parameter values.

### CODE REVIEW

File: frontend/src/metabase/api/parameters.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    return parametersApi.useSearchParameterValuesQuery(
+      {
+        ...expandParameters(params),
+        query: params.query,
+      },
+      options,
+    );
+  };
+
+function expandParameters({
```
Comment (ranquild) [inline frontend/src/metabase/api/parameters.ts:86]: The file should be just about plain API endpoints, no logic like this should be here

File: frontend/src/metabase/api/parameters.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
+import { getNonVirtualFields } from "metabase-lib/v1/parameters/utils/parameter-fields";
+import { normalizeParameter } from "metabase-lib/v1/parameters/utils/parameter-values";
+import type { Parameter, ParameterValues } from "metabase-types/api";
+
+import { Api, skipToken } from "./api";
+import { idTag } from "./tags";
+
+type NormalizedParameter = ReturnType<typeof normalizeParameter>;
+
+export type LoadParameterValuesParams = {
```
Comment (ranquild) [inline frontend/src/metabase/api/parameters.ts:10]: Please move types for requests to `metabase-types` and follow naming conventions from other endpoints

**Review by ranquild — COMMENTED** [review]
Inline comments

File: frontend/src/metabase/api/parameters.ts (hunk 1 above)
Comment (romeovs) [reply frontend/src/metabase/api/parameters.ts]: What would be a good place to put wrappers like this?

**Review by ranquild — APPROVED** [review]

**Pull request merged**